### PR TITLE
Use monotonic clock for timestamp deltas when available

### PR DESCRIPTION
**Pull request by djudd**
Check for [_POSIX_MONOTONIC_CLOCK](https://linux.die.net/man/3/clock_gettime) and, if present, use `clock_gettime` with `CLOCK_MONOTONIC` instead of using `gettimeofday`.

I've verified that tests pass and use the monotonic clock on both Ubuntu 20.04.2 and macOS 11.4 (which apparently grew support for `clock_gettime` recently). Haven't tested on any other systems but did verify the tests still pass and _don't_ use the monotonic clock if I munge the check.

I don't write much C, so please tell me what terrible things I've done and how to make them better.

### CODE REVIEW

File: ext/stackprof/stackprof.c [hunk 1]
```diff
@@ -558,25 +592,23 @@ stackprof_record_sample()
 void
 stackprof_record_gc_samples()
 {
-    int delta_to_first_unrecorded_gc_sample = 0;
-    int i;
+    int64_t delta_to_first_unrecorded_gc_sample = 0;
+    size_t i;
```
Comment (djudd) [inline ext/stackprof/stackprof.c:596]: This is an incidental change but made a compiler warning go away.

**Comment by djudd** [comment]
Since I'm not sure @tmm1 is actively maintaining this anymore, @tenderlove want to take a look?

**Comment by tenderlove** [comment]
It seems fine but I'll check more seriously (not from my phone lol) tomorrow. Thanks for sending this!! It makes sense to use the monotonic clock

**Comment by djudd** [comment]
@tenderlove did you get a chance to check more seriously?

**Comment by tenderlove** [comment]
@djudd sorry, I totally forgot.  Just reviewed now, and it looks great!

**Pull request merged**